linkerd/app/outbound/src/http/server.rs

```diff
@@ -142,38 +142,35 @@ impl errors::HttpRescue<Error> for ServerRescue {
     fn rescue(&self, error: Error) -> Result<errors::SyntheticHttpResponse> {
         use super::logical::policy::errors as policy;
 
+        // A profile configured request timeout was encountered.
         if errors::is_caused_by::<http::ResponseTimeoutError>(&*error) {
             return Ok(errors::SyntheticHttpResponse::gateway_timeout(error));
         }
+
+        // A request with a `l5d-require-id` header are dispatched to endpoints
+        // with a different identity.
         if errors::is_caused_by::<IdentityRequired>(&*error) {
             return Ok(errors::SyntheticHttpResponse::bad_gateway(error));
         }
 
-        if errors::is_caused_by::<errors::FailFastError>(&*error) {
-            return Ok(errors::SyntheticHttpResponse::gateway_timeout(error));
-        }
-        if errors::is_caused_by::<errors::LoadShedError>(&*error) {
+        // No available backend can be found for a request.
+        if errors::is_caused_by::<errors::FailFastError>(&*error)
+            || errors::is_caused_by::<errors::LoadShedError>(&*error)
+        {
```

Comment:
changing the status code returned by `FailFastError` broke a test, FYI: https://github.com/linkerd/linkerd2-proxy/actions/runs/4418928220/jobs/7746691581